Core/PARStore.m

```diff
@@ -1456,6 +1513,21 @@ - (NSData *)blobDataAtPath:(NSString *)path error:(NSError **)error;
     // otherwise blobs are stored in a special blob directory
     __block NSError *localError = nil;
     NSURL *fileURL = [[self blobDirectoryURL] URLByAppendingPathComponent:path];
+    
+    // check first for a tombstone, which indicates that the
+    // blob had been deleted and should be ignored
+    NSURL *tombstoneURL = [fileURL URLByAppendingPathExtension:TombstoneFileExtension];
+    if ([[NSFileManager defaultManager] fileExistsAtPath: tombstoneURL.path]) {
+        NSString *description = [NSString stringWithFormat:@"Blob at path '%@' has a tombstone, indicating that it has been deleted.", self.storeURL.path];
+        ErrorLog(@"%@", description);
+        if (error != NULL)
+            *error = [NSError errorWithObject:self code:__LINE__ localizedDescription:description underlyingError:nil];
+        
+        // TODO: should we attempt to clean up sync errors here, by deleting the data file if it still exists?
```

Comment:
Yes, good idea. May as well remove the file if the tombstone exists. Probably some earlier crash.